src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -47,63 +47,75 @@ final class CachedAppConfiguration {
 		deviceTimeCheck: DeviceTimeCheckProtocol? = nil,
 		configurationDidChange: (() -> Void)? = nil
 	) {
+		Log.debug("[App Config] CachedAppConfiguration init called", log: .localData)
+
 		self.client = client
 		self.store = store
 		self.configurationDidChange = configurationDidChange
 
 		self.deviceTimeCheck = deviceTimeCheck ?? DeviceTimeCheck(store: store)
 
-		guard shouldFetch() else { return }
 
-		// edge case: if no app config is cached, omit a potentially existing ETag to force fetch a new configuration
-		let etag = store.appConfig == nil ? nil : store.lastAppConfigETag
+		guard shouldFetch() else { return }
 
 		// check for updated or fetch initial app configuration
-		fetchConfig(with: etag)
+		fetchConfig(with: store.appConfigMetadata?.lastAppConfigETag)
 	}
 
 	private func fetchConfig(with etag: String?, completion: Completion? = nil) {
+		Log.debug("[App Config] fetchConfig called with etag:\(etag ?? "nil")", log: .localData)
+
 		client.fetchAppConfiguration(etag: etag) { [weak self] result in
 			guard let self = self else { return }
 
 			switch result.0 /* fyi, `result.1` would be the server time */{
 			case .success(let response):
-				self.store.lastAppConfigETag = response.eTag
-				self.store.appConfig = response.config
-				self.completeOnMain(completion: completion, result: .success(response.config))
+				let configMetadata = AppConfigMetadata(
+					lastAppConfigETag: response.eTag ?? "\"ReloadMe\"",
+					lastAppConfigFetch: Date(),
+					appConfig: response.config
+				)
+
+				// Skip processing of config if it didn't change.
+				guard self.store.appConfigMetadata?.lastAppConfigETag != configMetadata.lastAppConfigETag else {
+					Log.debug("[App Config] Skip processing app config, because it didn't change", log: .localData)
+					self.completeOnMain(completion: completion, result: .success(response.config))
+					return
+				}
 
-				// keep track of last successful fetch
-				self.store.lastAppConfigFetch = Date()
+				Log.debug("[App Config] Persist new app configuration", log: .localData)
+				self.store.appConfigMetadata = configMetadata
 
 				// update revokation list
-				let revokationList = self.store.appConfig?.revokationEtags ?? []
+				let revokationList = self.store.appConfigMetadata?.appConfig.revokationEtags ?? []
 				self.packageStore?.revokationList = revokationList // for future package-operations
 				// validate currently stored key packages
 				do {
 					try self.packageStore?.validateCachedKeyPackages(revokationList: revokationList)
 				} catch {
-					Log.error("Error while removing invalidated key packages.", log: .localData, error: error)
+					Log.error("[App Config] Error while removing invalidated key packages.", log: .localData, error: error)
```

Comment:
We could create custom OSLog keys for `.cache` and `.appConfig`!? This would allow nicer filtering for the different components.

Reply:
Added .appConfig and used it.